app/src/lib/app-state.ts

```diff
@@ -526,3 +529,10 @@ export interface ICompareToBranch {
  * An action to send to the application store to update the compare state
  */
 export type CompareAction = IViewHistory | ICompareToBranch
+
+export type SuccessfulMergeBannerState =
+  | {
+      currentBranch: string
+      theirBranch: string
+    }
+  | false
```

Comment:
:art: could we use `null` here to indicate the lack of successful merge banner state? It feels weird here to be mashing together an object with a flag.